code/modules/mob/living/simple_animal/hostile/hostile2.dm

```diff
@@ -169,6 +169,7 @@
 	LostTarget()
 
 /mob/living/simple_animal/hostile/proc/Goto(var/target, var/delay, var/minimum_distance)
+	step_towards(src, target) //weird but necessary so they try to bump openable obstacles
```

Comment:
Nope, get_step_towards2 looks for and returns free turfs on the path to target respecting dense objects though it does use a blunt get_step() if no free step is found. 

The issue is that step/walk_to, which Goto uses, does **not** try to path through dense objects to see if they can be opened, just gives up.

I could go and try to implement walk2 procs that use g_s_t2 (they're probably very simple loops of get_step and step) buuut it would be a bit of a bitch without much gain. Especially now that there's a check so a plain ol' step_towards won't override minimum distance.

For that matter, turns out we do have A* implemented. It just isn't used by simple_animals, but it is used by, say, Beepsky, which explains why he could open doors. TMYK. Maybe I'll just make a not_so_simple_animal class one day, would be cleaner to merge anyways...